hail/src/main/scala/is/hail/annotations/RegionPool.scala

```diff
@@ -96,8 +99,10 @@ final class RegionPool(strictMemoryCheck: Boolean = false) extends AutoCloseable
          |       used: ${ usedBlockCounts.mkString(", ") }""".stripMargin)
   }
 
+  override def finalize(): Unit = close()
+
   def close(): Unit = {
-    log.info(s"freeing RegionPool with $totalAllocatedBytes bytes allocated")
+    log.info(s"freeing RegionPool with $totalAllocatedBytes bytes (${ readableBytes(totalAllocatedBytes) } allocated")
```

Comment:
may as well include the thread id here too

Reply:
did this in #7076